pkg/expapi/validation/validation.go

```diff
@@ -156,3 +158,50 @@ func ValidateDaemonSpec(spec *expapi.DaemonSpec) errs.ValidationErrorList {
 func ValidateDaemonName(name string, prefix bool) (bool, string) {
 	return apivalidation.NameIsDNSSubdomain(name, prefix)
 }
+
+func ValidateJob(job *expapi.Job) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	// Jobs and rcs have the same name validation
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&job.ObjectMeta, true, apivalidation.ValidateReplicationControllerName).Prefix("metadata")...)
+	allErrs = append(allErrs, ValidateJobSpec(&job.Spec).Prefix("spec")...)
+	return allErrs
+}
+
+func ValidateJobSpec(spec *expapi.JobSpec) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+
+	if spec.Parallelism != nil && *spec.Parallelism < 0 {
+		allErrs = append(allErrs, errs.NewFieldInvalid("parallelism", spec.Parallelism, isNegativeErrorMsg))
+	}
+	if spec.Completions != nil && *spec.Completions < 0 {
+		allErrs = append(allErrs, errs.NewFieldInvalid("completions", spec.Completions, isNegativeErrorMsg))
+	}
+
+	selector := labels.Set(spec.Selector).AsSelector()
+	if selector.Empty() {
+		allErrs = append(allErrs, errs.NewFieldRequired("selector"))
```

Comment:
I guess you are making it as similar to replication controller as possible.  So, I guess that is fine.